mne/annotations.py

```diff
@@ -857,6 +857,36 @@ def _select_annotations_based_on_description(descriptions, event_id, regexp):
     return event_sel, event_id_
 
 
+def _select_events_based_on_id(events, event_desc):
+    """Get a collection of events and returns index of selected."""
+    event_desc_ = dict()
+    dropped = []
+    # Iterate over the sorted descriptions so that the Counter mapping
+    # is slightly less arbitrary
+    for e in events:
+        if e[2] in event_desc_:
+            continue
+
+        if isinstance(event_desc, dict):
+            if event_desc.get(e[2]) is not None:
+                event_desc_[e[2]] = event_desc[e[2]]
+            else:
+                continue
```

Comment:
No need for `continue` here / this branch, nothing happens after the conditionals anyway